### PR TITLE
feat: Add plugin info to analytics

### LINKED ISSUE #15882: [Task]: add plugin information to datasource save events
### Is there an existing issue for this?

- [X] I have searched the existing issues

### SubTasks

These two events:
SAVE_DATA_SOURCE
SAVE_DATA_SOURCE_CLICK

...we don't know what plugins were used in these datasources. Please add this information so that we can more easily track usage of this event broken down by plugin type.

### PR DESCRIPTION
**Pull request by Irongade**
This PR adds plugin information to  the SAVE_DATA_SOURCE and SAVE_DATA_SOURCE_CLICK events, so they can be observed.

Fixes #15882

- Chore (housekeeping or task changes that don't impact user perception)

### Dev activity
- [x] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

### QA activity:
- [ ] Test plan has been approved by relevant developers
- [ ] Test plan has been peer reviewed by QA
- [ ] Cypress test cases have been added and approved by either SDET or manual QA
- [ ] Organized project review call with relevant stakeholders after Round 1/2 of QA
- [ ] Added Test Plan Approved label after reveiwing all Cypress test

### CODE REVIEW

File: app/client/src/pages/Editor/DataSourceEditor/RestAPIDatasourceForm.tsx [hunk 1]
```diff
@@ -252,6 +255,8 @@ class DatasourceRestAPIEditor extends React.Component<
     AnalyticsUtil.logEvent("SAVE_DATA_SOURCE_CLICK", {
       pageId: this.props.pageId,
       appId: this.props.applicationId,
+      pluginName: this.props?.plugin?.name || "",
+      pluginPackageName: this?.props?.plugin?.packageName || "",
```
Comment (sneha122) [inline app/client/src/pages/Editor/DataSourceEditor/RestAPIDatasourceForm.tsx:259]: @Irongade Is there any advantage in adding pluginPackageName as well when we are adding plugin name as well?

Comment (Irongade) [reply app/client/src/pages/Editor/DataSourceEditor/RestAPIDatasourceForm.tsx:259]: I used the same format for other analytical events that sends plugin information.

**Review by sneha122 — DISMISSED** [review]

**Comment by Irongade** [comment]
@prapullac Kindly add test plan approved. It is an analytics PR and requires no QA checks.

**Comment by prapullac** [comment]
Based on the above comment adding the label

**Pull request merged**